src/main/java/backend/Logic.java

```diff
@@ -232,7 +232,13 @@ public void removeUnusedModels(Set<String> reposInUse) {
     }
 
     public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
-        uiManager.updateRateLimits(rateLimits);
+        uiManager.updateRateLimits(rateLimits, true);
+        return rateLimits;
+    }
+
+    public ImmutablePair<Integer, Long> updateRemainingRateWithoutUpdateRefreshRate(ImmutablePair<Integer,
+                                                                                    Long> rateLimits) {
```

Comment:
This `public` method's name is a bit confusing, better have a comment to explain its behaviour.